rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,211 @@
+
+Having Glides handle all trip assignments will require Glides having a direct feed of all vehicle position data. Currently, RTR excludes some known non-revenue yard pull-outs and pull-backs, which makes sense from a passenger information perspective. Internal consumers like Glides will need a more direct feed of the data before any filtering is applied. While the exact mechanism for achieving this is out of scope for this RFC, it is worth noting as a requirement as it relates to other discussions around re-architecting RTR. One possibility that has been discussed is splitting RTR into a data aggregation and state tracking component and a predictions component, set up such that other applications like Glides can also be consumers of the data from the former.
+
+## Additional RTR prediction considerations
+
+Much like other realtime prediction systems, RTR generates predictions not only for currently-operating trips but for trips that have not yet begun where we are already tracking a vehicle on an earlier trip in the same block. This has posed problems with light rail as those operations are generally less block-based due to factors like having many yards at terminal locations allowing for more ad-hoc swapping of vehicles between trips. When inspectors enter car numbers on future trips this can, however, create implicit block relationships in realtime. This RFC does not directly solve the issue of how RTR should link subsequent trips for prediction purposes. What it does do, however, is make the relationship between realtime and schedule data clearer via better trip assignments, so that RTR can (for example) more easily factor in runs for the purposes of trip linking based on TODS data, if that is deemed to be a desirable approach.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The primary drawback of doing this is if we decide that the benefits outlined in the [Motivation](#motivation) section are not worth the effort. There are also potential downsides to having Glides become the source of truth for trip assignment data; some of those are discussed in the [Rationale and alternatives](#rationale-and-alternatives) section below.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Glides versus RTR as the source of truth for assignments
+
+In deciding which system would be the source of truth for assignments between trains and trips, I had to weight the pros and cons of both RTR and Glides taking that logic on, ultimately landing on having it live in Glides. The following factors influenced my decision:
+
+1. Assigning trips to trains is a fundamentally stateful exercise that depends on a trains progress on its previous trip, location, trainsheet data about upcoming trips, and other factors. RTR has some stateful logic, but its state storage and management system is not very advanced. Glides has a relational database that is already updated with information about train movements and trips as they happen.
```

Comment:
thought: it also means there’s a person in the loop on the Glides side, making decisions about mapping vehicles to trips. that’s true on the bus and (to and extent) on the heavy rail side as well.